### PR TITLE
add flow chart and discussion on external data

### PR DESCRIPTION
**Pull request by f-hafner**
working on https://github.com/carpentries-incubator/lesson-R-packaging/issues/81: adding notes/exercises from [day 4](https://github.com/esciencecenter-digital-skills/2022-09-14-ds-rpackaging/blob/main/files/CollaborativeDocument-week4.md) on data.

### features
- add the flow chart in the beginning
- combine "Do you need raw data as part of your package? " (exercise from collaborative document) and note "note about raw data" (existing episode) into a discussion.

### open question

what to do with the part "Do you need data you can store in your package as an .Rda file?" ([link](https://github.com/esciencecenter-digital-skills/2022-09-14-ds-rpackaging/blob/main/files/CollaborativeDocument-week4.md))

**Proposal:** if we want to include it, we can create exercise "using data for example in documentation". ie, add `@example` to the documentation of `make_groups()`. The example first loads a vector of names and then calls the function. For this, we need to add the data, roughly following the content of the exercise from the collaborative document. 

*advantage:* hands-on, and one sees the purpose of having external data in action. adding an example fits nicely into the lesson after the episode on documenting the package.
*disadvantage:* Content overlaps with the current lesson material (step 1 and 2 in the episode). maybe creates confusion with the documentation episode?

any thoughts, @bvreede ?

### CODE REVIEW

**Comment by PabRod** [comment]
## TL;DR 
This is more than good enough. Feel free to merge and close. Thanks!

## Details
The flow chart is brilliant. And the addition of a subsection mentioning other types of data is a very good one.

Regarding the use of `@example`, I am not sure advanced documentation is a can of worms we want to open. The reason is exactly what you mention: overlap. And not only with the materials, but also with the concept of a package, where examples belong to vignettes and tests to... well, to tests!

Long story short, feel free to merge this. And also to discuss my point of view, of course :)

File: _episodes/09-data.md [hunk 1]
```diff
@@ -100,10 +117,29 @@ We'll save this text in `R/example_names.R`, and we are ready to go.
...
 
-## A note about raw data
 
-It should be noted that the way of dealing with data we just described will **only** work with _R_ data files, that is, those with `.rda` or `.RData` extensions.
+## Using data with extensions other than `.rda` or `.RData`
+
+> ## Discussion
+> When do you think is it useful for a package to include data that do not have the `.rda` or `.RData` extensions?
+> > ## Solution
+> > Having files without the _R_ extensions is when one of the main purposes of the package is to read external files. For instance, the  [readr](https://readr.tidyverse.org/) package loads rectangular data from files where the values are comma- or tab-separated.
```
Comment (bvreede) [inline _episodes/09-data.md:126]:
```suggestion
> > Having files without the _R_ extensions is useful when one of the main purposes of the package is to read external files. For instance, the  [readr](https://readr.tidyverse.org/) package loads rectangular data from files where the values are comma- or tab-separated. 
```

**Review by bvreede — APPROVED** [review]
